### PR TITLE
[coretext] Update for iOS 10 beta1

### PR DESCRIPTION
**Pull request by VincentDondain**
Here's the diff to explain why `HorizontalInVerticalForms` is an int (:

-extern const CFStringRef kCTVerticalFormsAttributeName CT_AVAILABLE(10_5, 4_3);
+CT_EXPORT const CFStringRef kCTVerticalFormsAttributeName CT_AVAILABLE(10_5, 4_3);

+    @const      kCTHorizontalInVerticalFormsAttributeName
+    @abstract   Setting text in tate-chu-yoko form (horizontal numerals in vertical text).
+    @discussion Value must be a CFNumberRef. Default is int value 0. A value of 1
+                to 4 indicates the number of digits or letters to set in horizontal
+                form. This is to apply the correct feature settings for the text.
+                This attribute only works when kCTVerticalFormsAttributeName is set to true.

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
Build success

File: src/CoreText/CTStringAttributes.cs [hunk 1]
```diff
@@ -193,6 +197,15 @@ public CTStringAttributes (NSDictionary dictionary)
 			set {Adapter.SetNativeValue (Dictionary, CTStringAttributeKey.ForegroundColor, value);}
 		}
 
+		[iOS (10,0)][Mac(10,12)]
+		public CGColor BackgroundColor {
+			get {
+				var h = CFDictionary.GetValue (Dictionary.Handle, CTStringAttributeKey.BackgroundColor.Handle);
```
Comment (spouliot) [inline src/CoreText/CTStringAttributes.cs]: note: unmanaged bindings needs unit tests.

In this case an older version of iOS will likely return `null` for `CTStringAttributeKey.BackgroundColor` so an `NullReferenceException` will be thrown here. Using `?.` (simpler) would likely solve this (and return `null`).

File: src/CoreText/CTStringAttributes.cs [hunk 2]
```diff
@@ -281,6 +294,12 @@ public CTStringAttributes (NSDictionary dictionary)
 			}
 		}
 
+		[iOS (10,0)][Mac(10,12)]
+		public int? HorizontalInVerticalForms {
+			get {return Adapter.GetInt32Value (Dictionary, CTStringAttributeKey.HorizontalInVerticalForms);}
```
Comment (spouliot) [inline src/CoreText/CTStringAttributes.cs]: Add a test case for make sure there's no `ArgumentNullException` thrown by `GetInt32Value` in old version of iOS where `CTStringAttributeKey.HorizontalInVerticalForms` could be `null`.

File: src/CoreText/CTStringAttributes.cs [hunk 3]
```diff
@@ -281,6 +294,12 @@ public CTStringAttributes (NSDictionary dictionary)
...
+			set {Adapter.SetValue (Dictionary, CTStringAttributeKey.HorizontalInVerticalForms, value);}
```
Comment (spouliot) [inline src/CoreText/CTStringAttributes.cs]: Same kind of test for `SetValue`

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by VincentDondain** [comment]
@spouliot
```
StringAttributesTests
	[PASS] StringAttributesTests.BackgroundColor
	[PASS] StringAttributesTests.HorizontalInVerticalForms
	[PASS] StringAttributesTests.SimpleValuesSet
StringAttributesTests : 5 ms
```

**Comment by spouliot** [comment]
@VincentDondain did you run this on an older iOS version ? that's where a NRE would happen (if the constants are null)

File: src/CoreText/CTStringAttributes.cs [hunk 4]
```diff
@@ -294,10 +301,17 @@ public CTStringAttributes (NSDictionary dictionary)
...
 		}
 
-		[iOS (10,0)][Mac(10,12)]
+		[iOS (10,0)][Mac (10,12)]
 		public int? HorizontalInVerticalForms {
-			get {return Adapter.GetInt32Value (Dictionary, CTStringAttributeKey.HorizontalInVerticalForms);}
-			set {Adapter.SetValue (Dictionary, CTStringAttributeKey.HorizontalInVerticalForms, value);}
+			get {
+				var x = CTStringAttributeKey.HorizontalInVerticalForms;
+				return x != null ? Adapter.GetInt32Value (Dictionary, x) : new int? ();
```
Comment (spouliot) [inline src/CoreText/CTStringAttributes.cs]: why `new int? ();` ? instead of returning `null` since it's a `int?` property ?

Comment (VincentDondain) [inline src/CoreText/CTStringAttributes.cs]: That's what I tried initially but it did not work, I got a "value can't be null" exception.

**Comment by xamarin-release-manager** [comment]
Build success

File: src/CoreText/CTStringAttributes.cs (hunk 4 above)
Comment (VincentDondain) [inline src/CoreText/CTStringAttributes.cs]: Never mind looks like it was an other issue... I just thought `new int? ();` fixed it. It's confusing, I really tried multiple times, likely didn't clean properly.
All good now 👍

**Comment by xamarin-release-manager** [comment]
Build success

**Pull request merged**